### PR TITLE
Update version, uses annotation and copyright year

### PR DESCRIPTION
**Pull request by casella**
This PR updates all version and uses annotation to 4.1.0. It also updates copyright dates to 2024. Finally, it removes a spurious version annotation in a sub-package.

### CODE REVIEW

File: Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo
Comment (maltelenz) [inline Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo:6]: I think this is wrong. It is a Release Notes item for 4.0.0, so the version and date shouldn't change?

File: Modelica/package.mo
Comment (maltelenz) [inline Modelica/package.mo:9084]: Was it intentional to push the `dateModified` _backwards_ in time, to January 2020?

File: ModelicaTest/package.mo
Comment (maltelenz) [inline ModelicaTest/package.mo:2]: I think we should just remove the " - Version 4.1.0". The information is readily available elsewhere, and it only adds to the maintenance burden.

File: Modelica/package.mo
Comment (maltelenz) [inline Modelica/package.mo:2]: I think we should just remove the " - Version 4.1.0". The information is readily available elsewhere, and it only adds to the maintenance burden.

File: ObsoleteModelica4.mo
Comment (maltelenz) [inline ObsoleteModelica4.mo:2887]: Is the next time we will have conversion scripts going to be 5.0.0 (according to semantic versioning)? I think so.

Will we at that point introduce an `ObsoleteModelica5.mo`? Probably?

I think this change is unnecessary and confusing. Same for the one on line 2900 below.

**Review by maltelenz — CHANGES_REQUESTED** [review]
Requesting changes according to comments.

**Review by beutlich — CHANGES_REQUESTED** [review]
I believe the copyright year in the license files

* LICENSE
* Modelica/Resources/Licenses/LICENSE_Modelica*.txt

also should get bumped.

**Review by beutlich — CHANGES_REQUESTED** [review]
The C library's version also needs to be bumped: Modelica/Resources/BuildProjects/autotools/configure.ac

File: ObsoleteModelica4.mo
Comment (henrikt-ma) [inline ObsoleteModelica4.mo:2900]: See @maltelenz's comment above.  I suggest simply dropping the version here:
```suggestion
Modelica.Math.Matrices.LAPACK.dgeqp3.
```
and additionally change line 2896 above like this:
```
\"ObsoleteModelica4.XXX\" should be renamed to \"Modelica.YYY\".
```
Note that there should be no need to point of the version of MSL used here, since ObsoleteModelica4 doesn't exist for versions prior to 4.

File: Modelica/Clocked/package.mo
Comment (HansOlsson) [inline Modelica/Clocked/package.mo:71]:
```suggestion
<tr><td> 2019 </td><td>The <em>Modelica_Synchronous</em> library is included as <em>Modelica.Clocked</em> in the Modelica Standard Library.</td></tr>
```
Just drop the version here, for two reasons:
- No need to update every time.
- It's a revision stating that this happened in 2019, clearly it wasn't added to 4.1.0 in 2019.

An alternative would be to replace it with "starting from 4.0.0" - but why bother?

File: Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo
Comment (HansOlsson) [reply Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo:6]: Agreed.
It also indicates that the idea of adding ReleaseNotes to sub-libraries in #2990 might not be a good idea, and to me it would be preferable if we had the following policy:
- Don't _add_ ReleaseNotes to sub-libraries.
- Keep but do not update ReleaseNotes. Ideally the last entry in a sub-package should be the statement that it was added to MSL.

Which means that it was an error to add this ReleaseNotes class.
I know that there are other opinions, but to me it seems to take too much effort to update the sub-package release notes, and thus it isn't done consistently. That means that it is not useful for figuring out what was changed.

Comment (HansOlsson) [inline Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo:6]:
```suggestion
<h5>Version 4.0.0, 2020-06-04</h5>
```
As indicated above.

**Comment by Harisankar-Allimangalath** [comment]
@beutlich @HansOlsson @maltelenz @henrikt-ma  can you please review the changes . Thankyou

File: ObsoleteModelica4.mo
Comment (HansOlsson) [inline ObsoleteModelica4.mo:2887]:
```suggestion
The conversion script for version 4.0.0 changes references in existing
```
The suggestion from @maltelenz

**Review by HansOlsson — CHANGES_REQUESTED** [review]
A number of minor changes have been requested.

**Comment by HansOlsson** [comment]
> @beutlich @HansOlsson @maltelenz @henrikt-ma can you please review the changes . Thankyou

I think it would be good to first address the current comments, and then ask for a review.

File: Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo
Comment (arunkumar-narasimhan) [reply Modelica/Electrical/Analog/UsersGuide/ReleaseNotes.mo]: Didn't see your changes here before committing the correction myself. @HansOlsson

File: Complex.mo
Comment (arunkumar-narasimhan) [inline Complex.mo:253]: Should the dateModified be updated to versionDate here as well?

File: ModelicaServices/package.mo
Comment (arunkumar-narasimhan) [inline ModelicaServices/package.mo:237]: Should the dateModified be same as versionDate here?

File: ModelicaTest/package.mo
Comment (arunkumar-narasimhan) [inline ModelicaTest/package.mo:56]: Should the dateModified be same as versionDate here?

File: ModelicaTestOverdetermined.mo
Comment (arunkumar-narasimhan) [inline ModelicaTestOverdetermined.mo:613]: Should the dateModified be same as versionDate here?

File: ObsoleteModelica4.mo
Comment (arunkumar-narasimhan) [inline ObsoleteModelica4.mo:2881]: Should the dateModified be same as versionDate here?

**Comment by Harisankar-Allimangalath** [comment]
@beutlich @HansOlsson @henrikt-ma @maltelenz  we have addressed the changes mentioned , can you please go through the changes. 

Thankyou

File: Modelica/Resources/C-Sources/ModelicaFFT.c
Comment (HansOlsson) [inline Modelica/Resources/C-Sources/ModelicaFFT.c:4]:
```suggestion
   Copyright (C) 2003-2010, Mark Borgerding
```
As far as I know there hasn't been any action from Mark; so don't update that line.

**Review by HansOlsson — APPROVED** [review]
Looks good.
Reverting the change for Mark, and deciding on dateModified isn't blocking

**Comment by Harisankar-Allimangalath** [comment]
@HansOlsson so once we get the approval and this is merged to the master , shall I  branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?

**Comment by HansOlsson** [comment]
> @HansOlsson so once we get the approval and this is merged to the master , shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?

Could _possibly_ do, but I would strongly prefer if #4256 #4189 and #4027 were also merged (and optionally #4204 ). And I assume others have similar considerations. It's a constant problem that we don't get the reviews.

**Comment by tobolar** [comment]
> And I assume others have similar considerations.

From my side, https://github.com/modelica/ModelicaStandardLibrary/pull/4229 (and #3406 but this is a too big for now :-( )

**Comment by HansOlsson** [comment]
> > @HansOlsson so once we get the approval and this is merged to the master , shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?
> 
> Could _possibly_ do, but I would strongly prefer if #4256 #4189 and #4027 were also merged (and optionally #4204 ). And I assume others have similar considerations. It's a constant problem that we don't get the reviews.

To clarify why I think this matters:
- After we have split off the maintenance branch it will be even more work to handle those PRs; with reviews for two different branches (master and maint). Thus by pushing work forward we are creating more work.

**Comment by Harisankar-Allimangalath** [comment]
> > > @HansOlsson so once we get the approval and this is merged to the master , shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?
> > 
> > 
> > Could _possibly_ do, but I would strongly prefer if #4256 #4189 and #4027 were also merged (and optionally #4204 ). And I assume others have similar considerations. It's a constant problem that we don't get the reviews.
> 
> To clarify why I think this matters:
> 
> * After we have split off the maintenance branch it will be even more work to handle those PRs; with reviews for two different branches (master and maint). Thus by pushing work forward we are creating more work.

We have pinged reviewers on those tickets, but not sure if we can expect a timely response. Should we proceed ahead with the branching to stick to the schedule or wait till Monday or at least till later today?

File: Modelica/Resources/C-Sources/stdint_wrap.h
Comment (beutlich) [inline Modelica/Resources/C-Sources/stdint_wrap.h:3]:
```suggestion
   Copyright (C) 2020-2024, Modelica Association and contributors
```

File: Modelica/package.mo
Comment (beutlich) [inline Modelica/package.mo:9084]:
```suggestion
dateModified = "2024-01-12 19:40:00Z",
```

Why a future date? This is promise unlikely to hold.

File: ModelicaReference/package.mo
Comment (beutlich) [inline ModelicaReference/package.mo:7314]:
```suggestion
  dateModified = "2024-01-12 19:40:00Z",
```

File: Modelica/package.mo
Comment (beutlich) [inline Modelica/package.mo:9083]:
```suggestion
versionDate="2024-01-12",
```

File: ModelicaReference/package.mo
Comment (beutlich) [inline ModelicaReference/package.mo:7313]:
```suggestion
  versionDate="2024-01-12",
```

File: ModelicaServices/package.mo
Comment (beutlich) [inline ModelicaServices/package.mo:236]:
```suggestion
    versionDate="2024-01-12",
```

File: ModelicaServices/package.mo
Comment (beutlich) [inline ModelicaServices/package.mo:237]:
```suggestion
    dateModified = "2024-01-12 19:40:00Z",
```

File: ModelicaTest/package.mo
Comment (beutlich) [inline ModelicaTest/package.mo:55]:
```suggestion
       versionDate="2024-01-12",
```

File: ModelicaTest/package.mo
Comment (beutlich) [inline ModelicaTest/package.mo:56]:
```suggestion
       dateModified = "2024-01-12 19:40:00Z",
```

File: ModelicaTestOverdetermined.mo
Comment (beutlich) [inline ModelicaTestOverdetermined.mo:612]:
```suggestion
       versionDate="2024-01-12",
```

File: ModelicaTestOverdetermined.mo
Comment (beutlich) [inline ModelicaTestOverdetermined.mo:613]:
```suggestion
       dateModified = "2024-01-12 19:40:00Z",
```

File: ObsoleteModelica4.mo
Comment (beutlich) [inline ObsoleteModelica4.mo:2880]:
```suggestion
              versionDate="2024-01-12",
```

File: ObsoleteModelica4.mo
Comment (beutlich) [inline ObsoleteModelica4.mo:2881]:
```suggestion
              dateModified = "2024-01-12 19:40:00Z",
```

File: Complex.mo
Comment (beutlich) [inline Complex.mo:252]:
```suggestion
versionDate="2024-01-12",
```

File: Complex.mo
Comment (beutlich) [inline Complex.mo:253]:
```suggestion
dateModified = "2024-01-12 19:40:00Z",
```

**Review by beutlich — CHANGES_REQUESTED** [review]

**Review by beutlich — CHANGES_REQUESTED** [review]
The copyright year of the LICENSE file (in repo root) should get bumped, too.

**Comment by beutlich** [comment]
> And I assume others have similar considerations.

Frome my side: #3700, #3999 (which works for me (and as workaround) but was rejected by review), #4250, #4251 and #4252.

**Comment by beutlich** [comment]
> shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?

What about the 40+ [open issues targeting the 4.1.0 milestone](https://github.com/modelica/ModelicaStandardLibrary/milestones/MSL4.1.0)?

**Comment by arunkumar-narasimhan** [comment]
> > shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?
> 
> What about the 40+ [open issues targeting the 4.1.0 milestone](https://github.com/modelica/ModelicaStandardLibrary/milestones/MSL4.1.0) (even the already approved ones)?

I have merged the approved ones, #3987 appears to be running checks at the moment, so will check back again.

**Comment by beutlich** [comment]
> > > shall I branch out maint/4.1.x and then tagout as v4.1.0-alpha.1?
> > 
> > 
> > What about the 40+ [open issues targeting the 4.1.0 milestone](https://github.com/modelica/ModelicaStandardLibrary/milestones/MSL4.1.0) (even the already approved ones)?
> 
> I have merged the approved ones, #3987 appears to be running checks at the moment, so will check back again.

No, the PR was older than 90 days, such that the (green) check results got removed by CI. It helps to rebase the branch to trigger a CI run again: -> Update with rebase

grafik

**Review by maltelenz — APPROVED** [review]
Looks good to me now.

File: Modelica/Resources/C-Sources/ModelicaFFT.h
Comment (beutlich) [inline Modelica/Resources/C-Sources/ModelicaFFT.h:4]:
```suggestion
   Copyright (C) 2003-2010, Mark Borgerding
```

**Review by beutlich — COMMENTED** [review]
Please squash these changes as a single commit. It is not useful to have 30+ commits on master for this PR:

**Review by beutlich — COMMENTED** [review]
Please squash these changes as a single commit. It is not useful to have 30+ commits on master for this PR. You can do in your git client manually or directly in GitHub wne slecting the proper merge strategy.

**Review by maltelenz — APPROVED** [review]
Still looks good to me.

File: Modelica/package.mo
Comment (HansOlsson) [inline Modelica/package.mo:8489]: The href and the corresponding anchor should be named 4_1_0 as well.

```suggestion
<tr><td><a href=\"modelica://Modelica.UsersGuide.ReleaseNotes.Version_4_1_0\">Version 4.1.0</a></td><td>March 15, 2024</td></tr>
```

**Comment by Harisankar-Allimangalath** [comment]
I have committed the suggestion from @HansOlsson and  @beutlich we can do the squash and Merge option at the github itself while merging the PR.

**Comment by Harisankar-Allimangalath** [comment]
@maltelenz @HansOlsson @beutlich @GallLeo  since we added one commit few minutes before we are in need of further approvals . can you please review the changes . Thankyou

**Comment by HansOlsson** [comment]
> @maltelenz @HansOlsson @beutlich @GallLeo since we added one commit few minutes before we are in need of further approvals . can you please review the changes . Thankyou

You have to update the release part as well for 4_1_0.

**Comment by HansOlsson** [comment]
> > @maltelenz @HansOlsson @beutlich @GallLeo since we added one commit few minutes before we are in need of further approvals . can you please review the changes . Thankyou
> 
> You have to update the release part as well for 4_1_0.

Specifically the following two should be Version_4_1_0 instead of Version_X_Y_0

https://github.com/modelica/ModelicaStandardLibrary/blob/dc6bac0b4d5331b9bc52315f1d4a346faa721fbc/Modelica/package.mo#L2428
https://github.com/modelica/ModelicaStandardLibrary/blob/dc6bac0b4d5331b9bc52315f1d4a346faa721fbc/Modelica/package.mo#L2509

**Comment by beutlich** [comment]
> You have to update the release part as well for 4_1_0.

Done in dedicated PR #4251.

**Comment by HansOlsson** [comment]
> > You have to update the release part as well for 4_1_0.
> 
> Done in dedicated PR #4251.

Oh, I didn't realize that it was that messy.

#4251 renames Version_X_Y_0 to Version_4_1_0.
This PR changes the href from Version_X_Y_0 to Version_4_1_0.

If both are merged close together in time I guess it will be ok.

File: Modelica/package.mo
Comment (HansOlsson) [inline Modelica/package.mo:8490]:
```suggestion
<td>Month D, 20YY</td></tr>
<tr><td><a href=\"modelica://Modelica.UsersGuide.ReleaseNotes.Version_4_1_0\">Version 4.1.0</a></td><td>March 15, 2024</td></tr>
```
To me the version X_Y_0 was just a placeholder, and shouldn't be included in an actual release.
I checked and 4.0.0 didn't have that placeholder.

 A more radical solution would be to go back to the style of 3.2.2 and remove this entire table (assuming that tools will show the package contents anyway).

Comment (arunkumar-narasimhan) [reply Modelica/package.mo:8490]: I will commit with the placeholder removed.

**Review by HansOlsson — APPROVED** [review]

**Comment by Harisankar-Allimangalath** [comment]
@casella @beutlich , so we are planning to move ahead and branch out the maintenance and further procedures today itself .Do you have any suggestions over that ?

Thankyou

**Pull request merged**
